website/src/controllers/payments/paypal.js

```diff
@@ -26,9 +26,10 @@ paypal.configure({
 });
 
 var parseErr = function(res, err){
-  //var error = err.response ? err.response.message || err.response.details[0].issue : err;
-  var error = JSON.stringify(err);
-  return res.json(400,{err:error});
+  var error = err.response ? err.response.error_description || err.response.error : 'Paypal returned an invalid error message.';
+  return res.json(400, {err: 'Paypal error: ' + error});
```

Comment:
Instead of "Paypal returned an invalid error message" I think it would be better to say just "PayPal reported an error". It's a simpler message and therefore might be less stressful to any users who see it. Also that and the "PayPal error" string should be translatable. Ask if you're not sure how to achieve that!

Please also add this string after the error message:
"Please email admin@habitica.com for assistance and paste that error message into your email."